app/android/src/uk/co/lutraconsulting/EXIFUtils.java

```diff
@@ -0,0 +1,126 @@
+        Log.d(TAG, "WriteExif: " + src + " to file: ");
+        InputStream in = null;
+        try {
+            ExifInterface exifInterface = new ExifInterface(src);
+            for (String key: attributes.keySet()) {
+                exifInterface.setAttribute(key, attributes.get(key));
+            }
+            exifInterface.saveAttributes();
+        } catch (IOException e) {
+            // Handle any errors
+            Log.d(TAG, "WriteExif ERROR: " + e.getMessage());
+        } finally {
+            if (in != null) {
+                try {
+                    in.close();
+                } catch (IOException ignored) {
+                }
+            }
+        }
+    }
+
+    public static HashMap<String, String> getExifGpsAttributes(String filepath) {
+        List<String> exifTags = new ArrayList<String>();
+        exifTags.add(GPS_BEARING_TAG);
+        exifTags.add(GPS_LAT_TAG);
+        exifTags.add(GPS_LON_TAG);
+        exifTags.add(GPS_DATE_TAG);
+
+        exifTags.add(GPS_PITCH_TAG);
+        exifTags.add(GPS_ROLL_TAG);
+        exifTags.add(CAMERA_ELEVATION_TAG);
+
+        return getEXIFdata(filepath, exifTags);
+    }
+
+    public static String getEXIFAttribute(String filepath, String tag) {
+        Log.d(TAG, "getEXIFAttribute: " + filepath + " - " + tag);
+        List<String> exifTags = new ArrayList<String>();
+        exifTags.add(tag);
+        return getEXIFdata(filepath, exifTags).get(tag);
+    }
+
+    public static HashMap<String, String> getEXIFdata(String filepath, List<String> exifTags) {
+        Log.d(TAG, "getEXIFdata: " + filepath + " to file: ");
+        InputStream in = null;
+        HashMap<String, String> result = new HashMap<String, String>();
+
+        if (exifTags.isEmpty()) return result;
+
+        try {
+            in = new FileInputStream(filepath);
+            ExifInterface exifInterface = new ExifInterface(in);
+            for (String tag : exifTags) {
+                String value = exifInterface.getAttribute(tag);
+                result.put(tag, value);
+                Log.d(TAG, "getEXIFdata: " + tag + " -> " + value);
+            }
+        } catch (IOException e) {
+            // Handle any errors
+        } finally {
+            if (in != null) {
+                try {
+                    in.close();
+                } catch (IOException ignored) {
+                }
+            }
+        }
+        return result;
+    }
+
+    // Converts double value to rational string representation
+    private static String calculateRational(double d) {
+
+        double precision_multiplier = Math.pow(10, DEGREE_PRECISION);
+        double rounded = Math.round(d * precision_multiplier)/precision_multiplier;
+        String[] fraction = Double.toString(rounded).split("\\.");
+
+        int denominator = (int)Math.pow(10, fraction[1].length());
+        int numerator = Integer.parseInt(fraction[0] + "" + fraction[1]);
+
+        return numerator + "/" + denominator;
+    }
```

Comment:
Hmm we probably don't really need to store angles as floating point values - the integer degree values should be just fine...